sunkit_image/granule.py

```diff
@@ -0,0 +1,289 @@
+"""
+This module contains functions that will segment images for granule detection.
+"""
+import logging
+
+import numpy as np
+import scipy.ndimage as sndi
+import skimage
+from sklearn.cluster import KMeans
+
+import sunpy
+import sunpy.map
+
+__all__ = [
+    "segment",
+    "get_threshold",
+    "trim_intergranules",
+    "mark_brightpoint",
+    "kmeans_segment",
+    "cross_correlation",
+]
+
+METHODS = ["li", "otsu", "isodata", "mean", "minimum", "yen", "triangle"]
+
+
+def segment(smap, resolution, *, skimage_method="li", mark_dim_centers=False):
+    """
+    Segment an optical image of the solar photosphere into tri-value maps with:
+
+     * 0 as intergranule
+     * 1 as granule
+     * 1.5 as brightpoint 
+
+    If mark_dim_centers is set to True, an additional label, 0.5, will be assigned to
+    dim grnanule centers. 
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        `~sunpy.map.GenericMap` containing data to segment.
+    resolution : `float`
+        Spatial resolution (arcsec/pixel) of the data.

```

Comment:
I wonder if this is something that we should get from the input map?

Reply:
We thought of doing that, but it would requre that the map has a valid header. While this might be a resonable assumption, some of our own data did not have all the information needed to read into a map header, so we had to use a dummy one. Perhaps we should indeed asssume that other users will be able to create maps containing accurate resolution values though.

Reply:
Ah so you created a dummy header to read in your files.

Hmm. My initial thought is that a map ideally should have a valid header and then you could check one of the CDELT keywords for the resolution. 

I will check to see if others agree with this first.

Reply:
The granule_testdata.fits file currently being used does have appropriate values for CDELT1 and CDELT2, so those could be read in directly from the header. 
This is all assuming square pixels of course...